morebits.js

```diff
@@ -2691,6 +2671,21 @@ Morebits.wiki.page = function(pageName, currentAction) {
 			// non-admin attempting to edit a protected page - this gives a friendlier message than the default
 			if (errorCode === 'protectedpage') {
 				ctx.statusElement.error('Failed to save edit: Page is protected');
+			// check for absuefilter hits: disallowed or warning
+			} else if (errorCode.indexOf('abusefilter') === 0) {
+				var desc = $(ctx.responseXML).find('error').find('abusefilter').attr('description');
+				if (errorCode === 'abusefilter-disallowed') {
+					ctx.statusElement.error('The edit was disallowed by the edit filter rule "' + desc + '".');
```

Comment:
Wouldn't `desc` be either the title of the filter, or the message it returns (doubtfully the latter since it's HTML)? In either case, you might want to add quotes around the message or append a colon: `'The edit was disallowed by the edit filter: ' + desc`. Otherwise it will read like the filter description is part of the sentence and be grammatically incorrect. It'd be really neat if we could link to the filter, too, but not a big deal if we can't or if it's too much work.

Reply:
Title, yeah, so 👍

I can play around with `id` once this is settled, see if we can't get something fancier in action.

Reply:
@MusikAnimal The language was mostly (entirely?) copied from the what previously existed, but warning and disallowed didn't match, so lmk how it looks now.